parsing/intermediate/dialects/defaultdialect/src/test/java/org/apache/calcite/test/DefaultDialectParserTest.java

```diff
@@ -27,4 +29,15 @@
   @Override protected SqlParserImplFactory parserImplFactory() {
     return DefaultDialectParserImpl.FACTORY;
   }
+
+  @Test void testCastFails() {
```

Comment:
nit: Should this be public? I've seen both public and private tests so I think both work but I'm just wondering if one option is better than the other.

Reply:
I am also interested to see if one option is better than the other. I have made this test to have package local access level because I migrated this test from SqlDialectParserTest.java and it was having package local access in that class. 

Thank you for leading my focus on this point. The class we are in, DeafultDialectParserTest, extends the SqlDialectParserTest. I figured out the way I named it as "testCastFails" will override the same name test in it's parent class. So I give it a different name in https://github.com/googleinterns/calcite/pull/216/commits/88c8d28bd18f4c2d271f4356c8b9967ef0064869.

Also, I don't find any different in behavior having it as package local or public. But it can't be private, because the parent class test has package local access level.